src/Core/Commit.js

```diff
@@ -0,0 +1,314 @@
+/**
+ * @param {*} element
+ * @return {Element}
+ */
+function commitElement (element) {
+	if (element != null)
+		switch (element.constructor) {
+			case Element:
+				return element
+			case Array:
+				return elementFragment(element)
+			case String:
+			case Number:
+				return elementText(element)
+			case Function:
+			case Promise:
+				return createElement(element)
+			case Boolean:
+			case Symbol:
+				break
+			default:
+				return elementUnknown(element)
+		}
+
+	return elementText('')
+}
+
+/**
+ * @param {Element} element
+ * @param {Element} snapshot
+ */
+function commitPromise (element, snapshot) {
+	snapshot.type.then(function (value) {
```

Comment:
should return promise for error handling or implement `.catch()`

Reply:
The Promise comes from the components render method returning it won't expose it to anyone that can use it, `.catch` can be implemented by the component, do you think we should also delegate errors in promises to the error boundary?

Reply:
Yes, uncaught promise errors in `render()` (and elsewhere) should trigger `componentDidThrow`